File: Fourth term/Fibers/Java/UnmanagedFiberAPI.java
package com.lab;

import com.sun.jna.Library;
import com.sun.jna.Native;

public class UnmanagedFiberAPI {

    public interface Kernel32 extends Library {
        int ConvertThreadToFiber(int lpParameter);

        void SwitchToFiber(int lpFiber);

        void DeleteFiber(int lpFiber);

        int CreateFiber(int dwStackSize, EventCallbackInterface lpStartAddress, int lpParameter);
    }

    public static Kernel32 kernel32 = (Kernel32) Native.loadLibrary("kernel32", Kernel32.class);

}

File: IgorKorobitsyn/Block1Task1/src/main/kotlin/UnmanagedFiberAPI.kt
@file:Suppress("FunctionName")

package fibers

import com.sun.jna.Library;
import com.sun.jna.Native;

object UnmanagedFiberAPI{

    public var kernel32 = Native.loadLibrary("kernel32", Kernel32::class.java) as Kernel32

    interface Kernel32: Library {

        fun ConvertThreadToFiber(lpParameter: Int): Int

        fun SwitchToFiber(lpFiber: Int)

        fun DeleteFiber(lpFiber: Int)

        fun CreateFiber(dwStackSize: Int, lpStartAddress: EventCallbackInterface, lpParameter: Int): Int
    }

    var count = 0
}